src/3d/TypeSizes.fst

```diff
@@ -433,12 +466,29 @@ let decl_size_with_alignment (env:env_t) (d:decl)
     | ExternFn _ _ _
     | ExternProbe _ -> d
 
+let idents_of_decl (d:decl) =
+  match d.d_decl.v with
+  | ModuleAbbrev i _
+  | Define i _ _ 
+  | TypeAbbrev _ i 
+  | Enum _ i _
+  | ExternFn i _ _
+  | ExternProbe i -> [i]
+  | Record names _ _ _
+  | CaseType names _ _
+  | OutputType { out_typ_names = names } 
+  | ExternType names -> [names.typedef_name; names.typedef_abbrev]
+
 let size_of_decls (genv:B.global_env) (senv:size_env) (ds:list decl) =
-  let env =
-    B.mk_env genv, senv in
-    // {senv with sizes = H.create 10} in
+  let env = B.mk_env genv, senv in
   let ds = List.map (decl_size_with_alignment env) ds in
+  let ge = B.global_env_of_env (fst env) in
+  ds |> List.iter (fun d ->

```

Comment:
Update the global environment with revised type declarations after fields (may) have been added for alignment padding